donjo-app/models/Sms_model.php

```diff
@@ -108,7 +108,8 @@ function list_data($o = 0, $offset = 0, $limit = 500)
 
 	public function insert_autoreply()
 	{
-		$data  = $_POST;
+		$data  = $this->input->post();
```

Comment:
Ini masih bentuk `mass assignment`, yaitu masih secara otomatis menyalin semua input ke data untuk disimpan. Ubah supaya `$data` diisi dengan mengambil kolom `post` satu-per-satu.  Lihat contoh di https://github.com/OpenSID/OpenSID/pull/3247/files

Lakukan perubahan ini untuk semua `mass assignment`, yaitu ubah semua assignment seperti `$data  = $this->input->post();`

Reply:
di view yang berhubungan yaitu `sms/setting` hanya ada satu field isian yaitu `autoreply_text`
data `autoreply_text` sudah divalidasi di baris 112
https://github.com/OpenSID/OpenSID/blob/2b0618b6407a5c67bc975f083c4974b0832ca0b7/donjo-app/models/Sms_model.php#L109-L112

untuk perbaikan PR ini, apa cukup hanya dengan menghapus baris 111 pak @eddieridwan ?

berikut ini view yang berhubungan:

https://github.com/OpenSID/OpenSID/blob/2b0618b6407a5c67bc975f083c4974b0832ca0b7/donjo-app/views/sms/setting.php#L10-L44

Reply:
Bisa langsung
`$data['autoreply_text'] = htmlentities($this->input->post('autoreply_text'));`

Tetapi solusinya secara umum adalah memindahkan input ke variable baru dan dari situ pindahkan satu-per-satu. 
```
		$post  = $this->input->post();
		$data['autoreply_text'] = htmlentities($post['autoreply_text']);
```
Atau kalau perlu dibersihkan dgn cara yg sama untuk insert dan update, lakukan di function terpisah seperti di pull request https://github.com/OpenSID/OpenSID/pull/3247/files
``` 
$data = $this->validasi($this->input->post();
```